src/tsconfig-parser.ts

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+import path from 'path';
+import ts from 'typescript';
+import { ParserOptions } from './temp-types-based-on-js-source';
+
+//------------------------------------------------------------------------------
+// Environment calculation
+//------------------------------------------------------------------------------
+
+/**
+ * Maps tsconfig paths to their corresponding file contents and resulting watches
+ * @type {Map<string, ts.WatchOfConfigFile<ts.SemanticDiagnosticsBuilderProgram>>}
+ */
+const knownWatchProgramMap = new Map<
+  string,
+  ts.WatchOfConfigFile<ts.SemanticDiagnosticsBuilderProgram>
+>();
+
+/**
+ * Maps file paths to their set of corresponding watch callbacks
+ * There may be more than one per file if a file is shared between projects
+ * @type {Map<string, ts.FileWatcherCallback>}
+ */
+const watchCallbackTrackingMap = new Map<string, ts.FileWatcherCallback>();
+
+/**
+ * Holds information about the file currently being linted
+ * @type {{code: string, filePath: string}}
+ */
+const currentLintOperationState = {
+  code: '',
+  filePath: ''
+};
+
+/**
+ * Appropriately report issues found when reading a config file
+ * @param {ts.Diagnostic} diagnostic The diagnostic raised when creating a program
+ * @returns {void}
+ */
+function diagnosticReporter(diagnostic: ts.Diagnostic): void {
+  throw new Error(
+    ts.flattenDiagnosticMessageText(diagnostic.messageText, ts.sys.newLine)
+  );
+}
+
+const noopFileWatcher = { close: () => {} };
+
+/**
+ * Calculate project environments using options provided by eslint and paths from config
+ * @param {string} code The code being linted
+ * @param {Object} options Options provided by ESLint core
```

Comment:
Did you mean to write ESLint here?

Reply:
Generalized all references I made to ESLint.